hs_collection_resource/views.py

```diff
@@ -0,0 +1,105 @@
+import logging
+
+from django.http import JsonResponse
+from django.db import transaction
+
+from hs_core.views.utils import authorize, ACTION_TO_AUTHORIZE
+from hs_core.hydroshare.utils import get_resource_by_shortkey, resource_modified
+
+logger = logging.getLogger(__name__)
+
+
+# update collection
+def update_collection(request, shortkey, *args, **kwargs):
+    """
+    Add resources to a collection. The POST request should contain a
+    list of resource ids for those resources to be part of the collection. Any existing resources
+    from the collection are removed before adding resources as specified by the list of
+    resource ids in the post request. Requesting user must at least have metadata view permission
+    for any new resources being added to the collection.
+
+    :param shortkey: id of the collection resource to which resources are to be added.
+    """
+
+    status = "success"
+    msg = ""
+    metadata_status = "Insufficient to make public"
+    try:
+        with transaction.atomic():
+            collection_res_obj, is_authorized, user = authorize(request, shortkey,
+                                                                needed_permission=ACTION_TO_AUTHORIZE.EDIT_RESOURCE)
+
+            if collection_res_obj.resource_type != "CollectionResource":
+                raise Exception("Resource {0} is not a collection resource.".format(shortkey))
+
+            # get res_id list from POST
+            updated_contained_res_id_list = request.POST.getlist("resource_id_list")
+
+            if len(updated_contained_res_id_list) > len(set(updated_contained_res_id_list)):
+                raise Exception("Duplicate resources were found for adding to the collection")
```

Comment:
@hyi You are right. The frontend already guaranteed there are no duplicate resources, so user should not be able to add duplicate resources into a collection from UI. The reason we added this check on the backend is mainly because later we may need to expose/wrap this backend view function as a rest api, at that time no UI can prevent duplicate resources from being added.